pcip/pcip-2.md

```diff
@@ -0,0 +1,621 @@
+     * @param replyFuture The future where the error will be reported.
+     */
+    void onSendRequestError(String correlationId, Throwable t, CompletableFuture<V> replyFuture);
+
+    /**
+     * Invoked after receiving a reply from the server successfully.
+     *
+     * <p>Please note that {@code replyFuture.complete(value)} must be executed at the end.
+     *
+     * @param correlationId The correlation ID associated with the reply.
+     * @param subscription The subscription name the reply was received on.
+     * @param value The value of the reply.
+     * @param replyFuture The future to be completed with the received value.
+     */
+    void onReplySuccess(String correlationId, String subscription, V value, CompletableFuture<V> replyFuture);
+
+    /**
+     * Invoked when an error occurs upon receiving a reply from the server.
+     *
+     * <p>Please note that {@code replyFuture.completeExceptionally(new Exception(errorMessage))} must be executed
+     * at the end.
+     *
+     * @param correlationId The correlation ID of the request.
+     * @param subscription The subscription name the error occurred on.
+     * @param errorMessage The error message associated with the reply.
+     * @param replyFuture The future to be completed exceptionally due to the error.
+     */
+    void onReplyError(String correlationId, String subscription, String errorMessage, CompletableFuture<V> replyFuture);
+
+    /**
+     * Invoked when receive reply message times out.
+     *
+     * @param correlationId The correlation ID associated with the request that timed out.
+     * @param t The timeout exception or relevant throwable.
+     */
+    void onTimeout(String correlationId, Throwable t);
+
+    /**
+     * Invoked when acknowledging reply message fails.
+     *
+     * <p>You can retry or record the messageId of the reply message for subsequent processing separately.
+     * <p>
+     *     This piece does not affect the current function. Because the reply message has been processed by
+     *     onReplySuccess or onReplyError. When the user-defined request success condition is met,
+     *     the user removes the request through the removeRequest method of rpc client.
+     *     Even if you receive the reply message corresponding to this request in the future.
+     *     But if there is no request in the pendingRequestMap, it will not be processed.
+     * </p>
+     *
+     * @param correlationId The correlation ID of the message.
+     * @param consumer The consumer that is acknowledging the message.
+     * @param msg The message that failed to be acknowledged.
+     * @param t The throwable error encountered during acknowledgment.
+     */
+    void onReplyMessageAckFailed(String correlationId, Consumer<V> consumer, Message<V> msg, Throwable t);
+}
+```
+
+### Configuration
+
+### CLI
+
+# Get started
+
+## Quick Start
+
+**Let's take a simple example to see how we can use the new SDK:**
```

Comment:
Good work! I have a small suggestion.
I think the steps in the example can be changed slightly to make it easier to understand.
1. Make a simple RPC service
2. Make an RPC client
3. Make some simple sample requests

Reply:
@liangyepianzhou I modified that part of the example, is there anything else that needs to be changed?

Reply:
Good work. Thanks for your contribution!